actors/builtin/market/market_balances.go

```diff
@@ -104,6 +104,29 @@ func (m *marketStateMutation) maybeLockBalance(addr addr.Address, amount abi.Tok
 	return nil, exitcode.Ok
 }
 
+func (m *marketStateMutation) removeAccountIfNoBalance(addr addr.Address) (error, exitcode.ExitCode) {
+	bal, err, code := getBalance(m.escrowTable, addr)
+	if err != nil {
+		return fmt.Errorf("failed to get escrow balance: %w", err), code
+	}
+
+	if bal.Equals(big.Zero()) {
+		prev, err := m.escrowTable.Remove(addr)
+		if err != nil {
+			return fmt.Errorf("failed to remove account: %w", err), exitcode.ErrIllegalState
+		}
+		AssertMsg(prev.Equals(big.Zero()), "previous escrow balance should be zero")
+
+		prev, err = m.lockedTable.Remove(addr)
+		if err != nil {
+			return fmt.Errorf("failed to remove locked account: %w", err), exitcode.ErrIllegalState
+		}
+		AssertMsg(prev.Equals(big.Zero()), "previous locked balance should be zero")
```

Comment:
I'm wondering (but not sure myself) whether it would be better to log this than assert it. The consequences of this assertion run in a CronTick could be dire, and, if the balance had been non-zero previously, it is zero now. What do you think?

Reply:
It doesn't seem wise to check the escrow table and then remove from a different table. The removal from each table should be guarded by a check on that table? If we expect them both to go at the same time, great, but we don't need to rely on it.

Reply:
This is done.